src/HomogeneousCalorimeter_geo.cpp

```diff
@@ -40,101 +34,14 @@ using namespace dd4hep;
- *   </detector>
- *
- *   <detector id="2" name="SomeBlocks" type="HomogeneousCalorimeter" readout="EcalHits">
- *     <position x="0" y="0" z="30*cm"/>
- *     <rotation x="0" y="0" z="0"/>
- *     <placements>
- *       <individuals sector="1"/>
- *         <module sizex="2.05*cm" sizey="2.05*cm" sizez="20*cm" vis="GreenVis" material="PbWO4"/>
- *         <wrapper thickness="0.015*cm" material="Epoxy" vis="WhiteVis"/>
- *         <placement x="1*cm" y="1*cm" z="0" id="1"/>
- *         <placement x="-1*cm" y="1*cm" z="0" id="2"/>
- *         <placement x="1*cm" y="-1*cm" z="0" id="3"/>
- *         <placement x="-1*cm" y="-1*cm" z="0" id="4"/>
- *       </individuals>
- *     </placements>
- *   </detector>
- *
- *   <detector id="2" name="DiskShapeCalorimeter" type="HomogeneousCalorimeter" readout="EcalHits">
- *     <position x="0" y="0" z="-30*cm"/>
- *     <rotation x="0" y="0" z="0"/>
- *     <placements>
- *       <disk rmin="25*cm" rmax="125*cm" length="20.5*cm" phimin="0" phimax="360*degree"
- *             envelope="false" sector="1">
- *         <module sizex="2.05*cm" sizey="2.05*cm" sizez="20*cm" vis="GreenVis" material="PbWO4"/>
- *         <wrapper thickness="0.015*cm" material="Epoxy" vis="WhiteVis"/>
- *       </disk>
- *     </placements>
- *   </detector>
- *
- *   <detector id="3" name="SomeLines" type="HomogeneousCalorimeter" readout="EcalHits">
- *     <position x="0" y="0" z="60*cm"/>
- *     <rotation x="0" y="0" z="0"/>
- *     <placements>
- *       <lines sector="1" mirrorx="true" mirrory="true"/>
- *         <module sizex="2.05*cm" sizey="2.05*cm" sizez="20*cm" vis="GreenVis" material="PbWO4"/>
- *         <wrapper thickness="0.015*cm" material="Epoxy" vis="WhiteVis"/>
- *         <line x="10.25*mm" y="10.25*mm" axis="x" begin="8" nmods="16"/>
- *         <line x="10.25*mm" y="30.75*mm" axis="y" begin="8" nmods="16"/>
- *         <line x="10.25*mm" y="51.25*mm" axis="z" begin="8" nmods="16"/>
- *       </individuals>
- *     </placements>
- *   </detector>
- * \endcode
- *
  * @{
  */
 
 // headers
-static std::tuple<int, int> add_individuals(Detector& desc, Assembly& env, xml::Collection_t& plm,
-                                            SensitiveDetector& sens, int id);
-static std::tuple<int, int> add_array(Detector& desc, Assembly& env, xml::Collection_t& plm, SensitiveDetector& sens,
-                                      int id);
-static std::tuple<int, int> add_disk(Detector& desc, Assembly& env, xml::Collection_t& plm, SensitiveDetector& sens,
-                                     int id);
-static std::tuple<int, int> add_12surface_disk(Detector& desc, Assembly& env, xml::Collection_t& plm,
+// static std::tuple<int, int> add_12surface_disk(Detector& desc, Assembly& env, xml::Collection_t& plm,
+//                                                SensitiveDetector& sens, int id);
+static std::tuple<int, std::pair<int, int>> add_12surface_disk(Detector& desc, Assembly& env, xml::Collection_t& plm,
```

Comment:
Tuple of int and pair of ints is kinda redundant. Why not a tuple of three ints?